### PR TITLE
support insert|replace grammar support #8377

### PR DESCRIPTION
**Pull request by dreamquster**
https://github.com/pingcap/tidb/issues/8377

### CODE REVIEW

**Comment by dreamquster** [comment]
@tiancaiamao PTAL

File: parser.y [hunk 1]
```diff
@@ -721,6 +721,7 @@ import (
 	PartitionDefinitionList 	"Partition definition list"
 	PartitionDefinitionListOpt	"Partition definition list option"
 	PartitionOpt			"Partition option"
+    PartitionNameOpt        "Partition name list option"
```
[the remark this answers is not in the document]
Comment (dreamquster) [reply parser.y]: How to set Ident in goland, how many spaces ident is equal to ?

[the remark this answers is not in the document]
Comment (kennytm) [reply parser.y]: @dreamquster the indentation should be 1 tab, not spaces

**Comment by tiancaiamao** [comment]
The problem is that if the parser PR is merged, while TiDB can't handle the new grammar, I'm not sure TiDB would throw a proper error or panic, so the integration testing is necessary. @dreamquster

File: parser.y (hunk 1 above)
[the remark this answers is not in the document]
Comment (zz-jason) [reply parser.y]: please use tab instead of spaces to indent the code.

**Comment by zz-jason** [comment]
Seems issue https://github.com/pingcap/tidb/issues/8377 is fixed by https://github.com/pingcap/tidb/pull/8411. @tiancaiamao Can we close this PR?

**Pull request closed without merging**